src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -2306,6 +2311,620 @@ internal unsafe void CM(B b)
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                // (6,33): error CS8759: Cannot bind function pointer to 'C.M()' because it is not a static method
+                //         delegate*<void> ptr1 = &M;
+                Diagnostic(ErrorCode.ERR_FuncPtrMethMustBeStatic, "M").WithArguments("C.M()").WithLocation(6, 33),
+                // (8,32): error CS8759: Cannot bind function pointer to 'int?.GetValueOrDefault()' because it is not a static method
+                //         delegate*<int> ptr2 = &i.GetValueOrDefault;
+                Diagnostic(ErrorCode.ERR_FuncPtrMethMustBeStatic, "i.GetValueOrDefault").WithArguments("int?.GetValueOrDefault()").WithLocation(8, 32)
+            );
+        }
+
+        [Fact]
+        public void AddressOf_MultipleInvalidOverloads()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+unsafe class C
+{
+    static int M(string s) => throw null;
+    static int M(ref int i) => throw null;
+
+    static void M1()
+    {
+        delegate*<int, int> ptr = &M;
+    }
+}");
+
+            comp.VerifyDiagnostics(
+                // (9,35): error CS8757: No overload for 'M' matches function pointer 'delegate*<int,int>'
+                //         delegate*<int, int> ptr = &M;
+                Diagnostic(ErrorCode.ERR_MethFuncPtrMismatch, "&M").WithArguments("M", "delegate*<int,int>").WithLocation(9, 35)
+            );
+        }
+
+        [Fact]
+        public void AddressOf_AmbiguousBestMethod()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+unsafe class C
+{
+    static void M(string s, object o) {}
+    static void M(object o, string s) {}
+    static void M1()
+    {
+        delegate*<string, string, void> ptr = &M;
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (8,48): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(string, object)' and 'C.M(object, string)'
+                //         delegate*<string, string, void> ptr = &M;
+                Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(string, object)", "C.M(object, string)").WithLocation(8, 48)
+            );
+        }
+
+        [Fact]
+        public void AddressOf_AsLvalue()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+unsafe class C
+{
+    static void M() {}
+    static void M1()
+    {
+        delegate*<void> ptr = &M;
+        &M = ptr;
+        M2(&M);
+        ref delegate*<void> ptr2 = ref &M;
+    }
+    static void M2(ref delegate*<void> ptr) {}
+}");
+
+            comp.VerifyDiagnostics(
+                // (8,9): error CS0131: The left-hand side of an assignment must be a variable, property or indexer
+                //         &M = ptr;
+                Diagnostic(ErrorCode.ERR_AssgLvalueExpected, "&M").WithLocation(8, 9),
+                // (9,12): error CS1503: Argument 1: cannot convert from '&method group' to 'ref delegate*<void>'
```

Comment:
>&method group [](start = 74, length = 13)

This looks confusing

Reply:
This is the same behavior we have today for method groups: https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQAdYwggdsgZgBoYQBLAGyIBMQBqAHwAEAmARgFgAoJggAlZ8AwnwDe3PpIH9qEChADmsCAIAsfACIAKAJQBuCVN5q+AWV1jDUqabZbT+q3wC+Tp8abrbWpGE19qHTFXLmcgA